AB_TESTING.md

```diff
@@ -0,0 +1,47 @@
+# A/B Testing
+
+## Intro
+
+Pathfora allows the user to perform A/B testing with the widgets.
+
+___
+
+Example file: `./examples/AB-testing example.html`
+
+> In the example file, we create two widgets, each in a separate testing group (groups are indexed starting from 0). As we open the page, a random value is generated into browser's cookies. According to this value, proper widgets are rendered on the page.
+
+___
+
+Pathfora provides currently two types of A/B testing:
+
+## Testing Modes
+
+### `100`
+
+This default mode renders all widgets regardless the user visiting the page.
+
+### `50/50`
+
+This mode divides users into two groups, each group with their separate set of widgets on the page.
+
+## Usage 
+
+In order to initialize A/B testing on the website, `setABTestingMode( mode )` has to be called before widgets are created on the page. For example:
+
+```js
+window.pathfora.setABTestingMode('50/50'); // Initialize a 50/50 AB testing
+window.pathfora.initializeWidgets(widgets);
+```
+
+When you initialize a mode in which widgets are displayed to users according to their testing group, you should then specify a `testGroup` parameter on the widgets (if the parameter is not defined, widget will always be visible):
+
+```js
+var messageBar = pathfora.Message({
+  id: 'asd',
+  layout: 'bar',
+  msg: 'Welcome to our website',
+  testGroup: 0
```

Comment:
@markhayden would you prefer it to be testing-oriented or widget-oriented then (both multiple groups on a single page and widget-to-test assignment) ? What I mean is:

Option 1 (widget-oriented approach, we assign tests to widgets same way we do now):
```js
var test1 = pathfora.ABTest('50/50');
var test2 = pathfora.ABTest('80/20');

var messageBar1 = pathfora.Message({
  id: 'a1',
  layout: 'bar',
  msg: 'test',
  testGroup: test1
});

var messageBar2 = pathfora.Message({
  id: 'a2',
  layout: 'bar',
  msg: 'test',
  testGroup: test2
});
```

Option 2 (test-oriented approach, we create tests from widgets):
```js

var messageBar1 = pathfora.Message({ id: 'a1' });
var messageBar2 = pathfora.Message({ id: 'a2' });

var test1 = pathfora.ABTest({
  type: '50/50',
  groups: [
    [ messageBar1 ],  // Group 1
    [ messageBar2 ]   // Group 2
  ]
});


var test2 = pathfora.ABTest({
  type: '80/20',
  groups: [
    [ messageBar1 ],  // Group 1
    [ messageBar2 ]   // Group 2
  ]
});
```